SharedContent/cs-winui-template/WinUI3TemplateCs/MainPage.xaml.cs

```diff
@@ -0,0 +1,156 @@
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace WinUI3TemplateCs
+{
+
+    public sealed partial class MainPage : Page
+    {
+        public static MainPage Current;
+
+        public List<Scenario> Scenarios
+        {
+            get { return this.scenarios; }
+        }
+
+        public MainPage()
+        {
+            InitializeComponent();
+
+            // This is a static public property that allows downstream pages to get a handle to the MainPage instance
+            // in order to call methods that are in this class.
+            Current = this;
+        }
+
+        public void NotifyUser(string strMessage, InfoBarSeverity severity, bool isOpen = true)
+        {
+            // If called from the UI thread, then update immediately.
+            // Otherwise, schedule a task on the UI thread to perform the update.
+            if (DispatcherQueue.HasThreadAccess)
+            {
+                UpdateStatus(strMessage, severity, isOpen);
+            }
+            else
+            {
+                DispatcherQueue.TryEnqueue(() =>
+                {
+                    UpdateStatus(strMessage, severity, isOpen);
+                });
+            }
+        }
+
+        private void UpdateStatus(string strMessage, InfoBarSeverity severity, bool isOpen)
+        {
+            infoBar.Message = strMessage;
+            infoBar.IsOpen = isOpen;
+            infoBar.Severity = severity;
+        }
+
+        private void NavView_Loaded(object sender, RoutedEventArgs e)
+        {
+            foreach (var item in scenarios)
+            {
+                NavView.MenuItems.Add(new NavigationViewItem
+                {
+                    Content = item.Title,
+                    Tag = item.ClassName,
+                    Icon = new FontIcon() { FontFamily = new("Segoe MDL2 Assets"), Glyph = "\uE82D" }
+                });
+            }
+
+            // NavView doesn't load any page by default, so load home page.
+            NavView.SelectedItem = NavView.MenuItems[0];
+
+            // If navigation occurs on SelectionChanged, this isn't needed.
+            // Because we use ItemInvoked to navigate, we need to call Navigate
+            // here to load the home page.
+            if (scenarios is not null && scenarios.Count > 0)
+            {
+                NavView_Navigate(scenarios.First().ClassName, new Microsoft.UI.Xaml.Media.Animation.EntranceNavigationTransitionInfo());
+            }
+        }
+
+        private void NavView_Navigate(string navItemTag, Microsoft.UI.Xaml.Media.Animation.NavigationTransitionInfo transitionInfo)
+        {
+            Type _page = null;
+
+            if (navItemTag == nameof(SettingsPage))
+            {
+                _page = typeof(SettingsPage);
+            }
+            else
+            {
+                var item = scenarios.FirstOrDefault(p => p.ClassName.Equals(navItemTag));
+                _page = Type.GetType(item.ClassName);
```

Comment:
You are dereferencing `item` without a null check, but you just used `FirstOrDefault` which explicitly allows for returning `null`. This is a `NullReferenceException` risk. If you really expect a match on the previous line, use `First` instead so it throws a more diagnosable exception rather than returning null. Otherwise add a null check.

This is the sort of bug that can be avoided by adding `<Nullable>enable</Nullable>` to the project file.

Reply:
Since all samples will have at least 1 scenario, changing from `FirstOrDefault` to `First`